### DISCUSSION TITLE
Issues with latin1/latin1_general_ci on Mysql8

### DISCUSSION THREAD

**Discussion by davidjohnsonmedaxion** [nextgenhealthcare/connect#6377] [General]
Hello,

Some quick context:
- We are running Mirth Connect Server 3.10.1 on Java version: 17.0.6
- We have a MySQL 5.7 backend hosted on AWS RDS.
- We are running in Docker, with the official Nextgen image: https://hub.docker.com/r/nextgenhealthcare/connect, at this SHA: 54ea2d97513cf46c521b23bb0022905585acd1746dffa49c76 e5be6adaf40409.
- We are running the community version without a paid license.
- Our default charset/collation is latin1/latin1_general_ci for all databases in our MySQL instance.

The above setup is currently working fine.

We need to migrate to RDS MySQL 8.4, however we are running into encoding issue when we do so. During a test run, upgrading from MySQL 5.7 to MySql 8.4, our channels were unable to read content which was encoded latin1/latin1_general_ci, when previously on MySQL 5.7 they were able to do. We were unable to find a combination of channel settings which fixed the issue, however manually changing the encoding on a given channel's tables to be utf8mb4, fixed the issue. It should be noted that during the test upgrade nothing about the channel configuration was changed.

It seems there is an issue for us in the combination of Mirth 3.10.1/Mysql8.4/latin1.

Has anyone heard of this problem before, and if so is there a work around? Does Mirth not play nice with this combination: ( Mirth 3.10.1/Mysql8.4/latin1 ) in general?

Is this something potentially mitigated by upgrading our version of Mirth, since we are a major version behind?

Note we are considering changing the encoding for database, and then trying a MySQL upgrade, but that is quite an undertaking given the amount of data we have in place.

Any help appreciated.

Cheers!

Also posted here: https://forums.mirthproject.io/forum/mirth-connect/support/185932-issues-with-latin1-latin1_general_ci-on-mysql8

**Comment by kirbykn2** [comment]
" It seems there is an issue for us in the combination of Mirth
3.10.1/Mysql8.4/latin1. "

Have you tried the combination of Mirth 4.5.x/Mysql8.4/latin1?


On Thu, Dec 12, 2024 at 12:59 PM David Johnson ***@***.***>
wrote:

> Hello,
>
> Some quick context:
>
>    - We are running Mirth Connect Server 3.10.1 on Java version: 17.0.6
>    - We have a MySQL 5.7 backend hosted on AWS RDS.
>    - We are running in Docker, with the official Nextgen image:
>    https://hub.docker.com/r/nextgenhealthcare/connect, at this SHA:
>    54ea2d97513cf46c521b23bb0022905585acd1746dffa49c76 e5be6adaf40409.
>    - We are running the community version without a paid license.
>    - Our default charset/collation is latin1/latin1_general_ci for all
>    databases in our MySQL instance.
>
> The above setup is currently working fine.
>
> We need to migrate to RDS MySQL 8.4, however we are running into encoding
> issue when do so. During a test run, upgrading from MySQL 5.7 to MySql 8.4,
> our channels were unable to read content which was encoded
> latin1/latin1_general_ci, when previously on MySQL 5.7 they were able to
> do. We were unable to find a combination of channel settings which fixed
> the issue, however manually changing the encoding on a given channel's
> tables to be utf8mb4, fixed the issue. It should be noted that during the
> test upgrade nothing about the channel configuration was changed.
>
> It seems there is an issue for us in the combination of Mirth
> 3.10.1/Mysql8.4/latin1.
>
> Has anyone heard of this problem before, and if so is there a work around?
> Does Mirth not play nice with this combination: ( Mirth
> 3.10.1/Mysql8.4/latin1 ) in general?
>
> Is this something potentially mitigated by upgrading our version of Mirth,
> since we are a major version behind?
>
> Note we are considering changing the encoding for database, and then
> trying a MySQL upgrade, but that is quite an undertaking given the amount
> of data we have in place.
>
> Any help appreciated.
>
> Cheers!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6377>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWDYSZVUD5ESPU5LBA3D2FHFIRAVCNFSM6AAAAABTQLAPW6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGY3DMNJUGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by davidjohnsonmedaxion** [reply]
We have not tried upgrading Mirth yet, but we are open to doing so.

**Comment by pacmano1** [comment]
@davidjohnsonmedaxion cross reference your forum post please.

**Reply by davidjohnsonmedaxion** [reply]
Done.
